app/models/type.rb

```diff
@@ -72,9 +72,9 @@ def <=>(type)
     name <=> type.name
   end
 
-  def self.all
-    find(:all, order: 'position')
-  end
+  # def self.all
+  #  find(:all, order: 'position')
+  # end
```

Comment:
@NobodysNightmare this results in a stack overflow in Rails 4 (see commit message 53ecf98b58d953051a924c7ee926bf8bf653d239). I'll check to see if using `ActiveRecord::QueryMethods` methods works though (i.e. `order('position')`). We could also set `default_scope`.